src/git.rs

```diff
@@ -11,8 +12,54 @@ pub fn git_fetch(root: &Path) -> Result<(), git2::Error> {
     Ok(())
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct BranchInfo {
+    name: String,
+    commit_hash: String,
+    time: i64,
+    message: Vec<String>,
+}
+
+fn get_branch_info(
+    branch_struct: Result<(Branch, BranchType), git2::Error>,
+) -> Result<BranchInfo, git2::Error> {
+    let branch = branch_struct?.0;
+    let lossy_name = String::from_utf8_lossy(branch.name_bytes()?);
+    let name = lossy_name
+        .strip_prefix("origin/")
+        .unwrap_or(&lossy_name)
+        .to_owned();
+
+    let branch_commit = branch.into_reference().peel_to_commit()?;
+    let message: Vec<String> = String::from_utf8_lossy(branch_commit.message_bytes())
+        .split_terminator("\n")
+        .map(String::from)
+        .collect();
+
+    Ok(BranchInfo {
+        name,
+        commit_hash: branch_commit.id().to_string(),
+        time: branch_commit.time().seconds(),
+        message,
+    })
+}
+
+pub fn git_list_branches(root: &Path) -> Result<Vec<BranchInfo>, git2::Error> {
+    let repo = Repository::open(root)?;
+    let git_branches: Result<Vec<BranchInfo>, git2::Error> = repo
+        .branches(Some(BranchType::Remote))?
+        // first branch seems to be HEAD, we don't want to display that to the
+        // users so skip it
+        .skip(1)
```

Comment:
It seems preferable to filter by checking the value rather than by position. Otherwise you’re dependent on what could be an implementation detail.

Reply:
yep thats better, have done it here instead, it was just easier to insert https://github.com/mrc-ide/outpack_server/pull/64/commits/b4c201a9e387ff391794c187ec148854debd5e24

Reply:
Sorry to be a bit nitpicky but it seems more appropriate to fix this here rather than in the next PR. I've double checked and it's more than just an implementation detail.

The result is actually sorted alphabetically, which means HEAD is generally first only because branch names are usually lowercase. If you have a branch that starts with a uppercase character in the A-G range it will get filtered out in place of HEAD.

Reply:
no worries, that makes sense, ive copied the filter implementation from the next PR into here where i filter by name